iceoryx_hoofs/vocabulary/include/iox/expected.hpp

```diff
@@ -14,27 +14,32 @@
 // limitations under the License.
 //
 // SPDX-License-Identifier: Apache-2.0
-#ifndef IOX_HOOFS_CXX_EXPECTED_HPP
-#define IOX_HOOFS_CXX_EXPECTED_HPP
+#ifndef IOX_HOOFS_VOCABULARY_EXPECTED_HPP
+#define IOX_HOOFS_VOCABULARY_EXPECTED_HPP
 
 #include "iceoryx_hoofs/cxx/attributes.hpp"
 #include "iceoryx_hoofs/cxx/functional_interface.hpp"
 #include "iceoryx_hoofs/cxx/helplets.hpp"
-#include "iceoryx_hoofs/cxx/optional.hpp"
-#include "iceoryx_hoofs/cxx/variant.hpp"
+#include "iox/optional.hpp"
+#include "iox/variant.hpp"
 
 #include <utility>
 
 namespace iox
 {
-namespace cxx
+namespace internal
 {
+/// @brief Type trait which verifies whether the passed type T is of type optional
+template <typename...>
+struct IsOptional;
+} // namespace internal
```

Comment:
Why is this needed?

Reply:
Good catch. I think in the meantime this was removed or renamed to `is_optional` on `master`. Removed it.